### DISCUSSION TITLE
@mantine/form devtools?

### DISCUSSION THREAD

**Discussion by vipindigiqt** [mantinedev/mantine#5157] [Feature requests]
Do you think we should have a @mantine/form devtools?

**Comment by rtivital** [comment]
It would be nice to have, but I do not really have the capacity to work on this any time soon.

**Reply by leomerida15** [reply]
I created a package that is a devtools for useForm. I hope it helps you and if you like it, we can add it to the maintenance documentation.

https://www.npmjs.com/package/mantine-form-devtools

**Comment by chewhx** [comment]
This is a neat idea. 
Usually we need to console log the values or render them to DOM, in order to find out what the current values are and what are the errors.
Is possible to write a custom component to be used within a page. 


```tsx
<MantineFormDevTool form={form} />
```

I made something like this as a concept. It works by passing the `form` object to a custom component using Affix, Drawer, Tabs.
I'm interested to explore into something that might work like react-query devtools. 
Instead of having to include the component on every page. We should include it at the top of the application and it can detect form context to present a panel.

Interested to hear ideas or opinions on this.

![Vite _ React _ TS](https://github.com/mantinedev/mantine/assets/5371541/05ac76be-1142-4f85-9d4c-f71d3a711c52)

**Reply by leomerida15** [reply]
I am very interested, and if you agree, we could explore the possibility of packaging it and releasing it for the community.

**Reply by leomerida15** [reply]
I created a version of the devtools for real use here attached as seen, I'm working on packaging it for npm

***dashboards***
***expan***
<img width="362" height="584" alt="image" src="https://github.com/user-attachments/assets/8a1e8e9a-6cc8-4b97-8cf0-7b98db5c39a5" />

***collapse***
<img width="381" height="213" alt="image" src="https://github.com/user-attachments/assets/c5ae2c74-b8e3-43d6-9027-1561a207dd57" />

<img width="1839" height="692" alt="image" src="https://github.com/user-attachments/assets/462f29f8-3055-4de6-9289-7333c5550d8c" />

***code use***


- with hook
<img width="821" height="538" alt="image" src="https://github.com/user-attachments/assets/5c9f3db8-a5e1-4601-8819-755614700517" />
- with tag element
<img width="799" height="33" alt="image" src="https://github.com/user-attachments/assets/c070ec45-6def-444f-9773-761dfdc70348" />
